modules/search/class.jetpack-search.php

```diff
@@ -429,6 +429,65 @@ public function search( array $es_args ) {
 		return $response;
 	}
 
+	/**
+	 * Displays HTML about the result of the spelling auto-correction.
+	 * Should be called during loop_start to be displayed below page headers and above post results.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $spell_text The text to show in the suggest block
+	 */
+	public function display_suggest_result( $spell_text ) {
+		printf( "<div class=\"suggest-result page-header\"><h4>" . $spell_text . "<br><br></h4></div>" );
+	}
+
+	/**
+	 * Displays HTML for "Did you mean, $suggestion?", which $suggestion being a link to search for the suggestion.
+	 * Should be called during loop_start to be displayed below page headers and above post results.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Query $query             The WP_Query object.
+	 */
+	public function filter__show_did_you_mean( $query ) {
+		if ( ! is_array( $this->search_result ) ) {
+			return;
+		}
+		if ( empty( $this->search_result['suggest'] ) ) {
+			return;
+		}
+		$suggestion = $this->search_result['suggest']['experimental-phrase-suggester'][0]['options'][0]['text'];
+		if ( $suggestion == $query->get( 's' ) ) {
+			return;
+		}
+		$suggestion_sentence = sprintf( wp_kses( _x( 'Did you mean, <em><a href="/?s=%1$s">%2$s?</a></em>', 'displayed as a spelling suggestion for a search query', 'jetpack' ) ), $suggestion, $suggestion );
```

Comment:
I think the URL here for this link needs to be based on whatever the current search URL is. For instance if the user is doing any filtering, or the user has selected a non-default sorting.